solcjs-get

```diff
@@ -25,51 +54,83 @@ function getVersionList (cb) {
   });
 }
 
-function downloadBinary (outputName, version, expectedHash) {
+function downloadBinary (version, cb) {
   console.log('Downloading version', version);
+  var targetPath = path.join(compilerDir, version);
 
-  // Remove if existing
-  if (fs.existsSync(outputName)) {
-    fs.unlinkSync(outputName);
-  }
-
-  process.on('SIGINT', function () {
-    console.log('Interrupted, removing file.');
-    fs.unlinkSync(outputName);
-    process.exit(1);
-  });
-
-  var file = fs.createWriteStream(outputName, { encoding: 'binary' });
   https.get('https://solc-bin.ethereum.org/bin/' + version, function (response) {
     if (response.statusCode !== 200) {
       console.log('Error downloading file: ' + response.statusCode);
       process.exit(1);
     }
+
+    fs.ensureDirSync(compilerDir);
+
+    var file = fs.createWriteStream(targetPath);
+
     response.pipe(file);
     file.on('finish', function () {
       file.close(function () {
-        var hash = '0x' + keccak256(fs.readFileSync(outputName, { encoding: 'binary' }));
-        if (expectedHash !== hash) {
-          console.log('Hash mismatch: ' + expectedHash + ' vs ' + hash);
-          process.exit(1);
-        }
         console.log('Done.');
+        if (cb) cb(targetPath);
       });
     });
   });
 }
 
-console.log('Downloading correct solidity binary...');
-
-getVersionList(function (list) {
-  list = JSON.parse(list);
-  var wanted = pkg.version.match(/^(\d+\.\d+\.\d+)$/)[1];
-  var releaseFileName = list.releases[wanted];
-  var expectedFile = list.builds.filter(function (entry) { return entry.path === releaseFileName; })[0];
-  if (!expectedFile) {
-    console.log('Version list is invalid or corrupted?');
-    process.exit(1);
-  }
-  var expectedHash = expectedFile.keccak256;
-  downloadBinary('soljson.js', releaseFileName, expectedHash);
-});
+if (argv.list) {
+  console.log('Getting the list of all versions ...');
+  getVersionList(function (list) {
+    list = JSON.parse(list).builds;
+    for (var i = list.length - 1; i >= 0; i--) {
+      console.log(list[i].version, list[i].path);
+    }
+    process.exit(0);
+  });
+} else if (argv.clean) {
+  console.log('Removing all local compilers in ' + compilerDir + ' ...');
+  fs.remove(compilerDir, function (err) {
+    if (err) return console.error(err);
+
+    console.log('Success! Cleaned ' + compilerDir);
+    process.exit(0);
+  });
```

Comment:
Not sure if it's a good idea to remove the whole dir. As a user I think I'd expect this to be more intelligent and only remove the binaries (otherwise why even provide an option for something that can be easily done with `rm`?). I think it would be safer to only remove files matching the valid name regex.